### ISSUE TITLE
Playing media with enqueue set to True is deprecated. Use 'add' instead

### ISSUE THREAD

**Issue by Mariusthvdb** [music-assistant/hass-music-assistant#271]
Experiencing some spontaneous restarts, not sure, but happened after clicking Play on a Spotify album in Music Assistant.
please have a look if this is Mass? or HA...

HA 2022.7.0.dev20220531

```
2022-05-31 13:38:00 ERROR (MainThread) [homeassistant.core] Error executing service: <ServiceCall media_player.play_media (c:01G4CYGSYZJPGZXXW73QB2FYAP): entity_id=['media_player.googlehome_library'], extra=enqueue=True, media_type=music, media_id=http://192.168.1.71:8095/media_player.googlehome_library.flac, enqueue=add>
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/core.py", line 1724, in catch_exceptions
    await coro_or_task
  File "/usr/src/homeassistant/homeassistant/core.py", line 1743, in _execute_service
    await cast(Callable[[ServiceCall], Awaitable[None]], handler.job.target)(
  File "/usr/src/homeassistant/homeassistant/helpers/entity_component.py", line 204, in handle_service
    await service.entity_service_call(
  File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 680, in entity_service_call
    future.result()  # pop exception if have
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 964, in async_request_call
    await coro
  File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 717, in _handle_entity_call
    await result
  File "/usr/src/homeassistant/homeassistant/components/cast/media_player.py", line 717, in async_play_media
    await self.hass.async_add_executor_job(
  File "/usr/local/lib/python3.9/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/usr/local/lib/python3.9/site-packages/pychromecast/quick_play.py", line 77, in quick_play
    controller.quick_play(**data)
  File "/usr/local/lib/python3.9/site-packages/pychromecast/controllers/media.py", line 544, in quick_play
    raise PyChromecastError()
pychromecast.error.PyChromecastError
2022-05-31 13:38:00 ERROR (MainThread) [homeassistant.core] Error executing service: <ServiceCall media_player.play_media (c:01G4CYGT288NBWWASWVVH4KEJ7): entity_id=['media_player.googlehome_library'], extra=enqueue=True, media_type=music, media_id=http://192.168.1.71:8095/media_player.googlehome_library.flac, enqueue=add>
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/core.py", line 1724, in catch_exceptions
    await coro_or_task
  File "/usr/src/homeassistant/homeassistant/core.py", line 1743, in _execute_service
    await cast(Callable[[ServiceCall], Awaitable[None]], handler.job.target)(
  File "/usr/src/homeassistant/homeassistant/helpers/entity_component.py", line 204, in handle_service
    await service.entity_service_call(
  File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 680, in entity_service_call
    future.result()  # pop exception if have
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 964, in async_request_call
    await coro
  File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 717, in _handle_entity_call
    await result
  File "/usr/src/homeassistant/homeassistant/components/cast/media_player.py", line 717, in async_play_media
    await self.hass.async_add_executor_job(
  File "/usr/local/lib/python3.9/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/usr/local/lib/python3.9/site-packages/pychromecast/quick_play.py", line 77, in quick_play
    controller.quick_play(**data)
  File "/usr/local/lib/python3.9/site-packages/pychromecast/controllers/media.py", line 544, in quick_play
    raise PyChromecastError()
pychromecast.error.PyChromecastError
2022-05-31 13:38:00 ERROR (MainThread) [homeassistant.core] Error executing service: <ServiceCall media_player.play_media (c:01G4CYGT5EABJEHENV8TZEKNX7): entity_id=['media_player.googlehome_library'], extra=enqueue=True, media_type=music, media_id=http://192.168.1.71:8095/media_player.googlehome_library.flac, enqueue=add>
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/core.py", line 1724, in catch_exceptions
    await coro_or_task
  File "/usr/src/homeassistant/homeassistant/core.py", line 1743, in _execute_service
    await cast(Callable[[ServiceCall], Awaitable[None]], handler.job.target)(
  File "/usr/src/homeassistant/homeassistant/helpers/entity_component.py", line 204, in handle_service
    await service.entity_service_call(
  File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 680, in entity_service_call
    future.result()  # pop exception if have
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 964, in async_request_call
    await coro
  File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 717, in _handle_entity_call
    await result
  File "/usr/src/homeassistant/homeassistant/components/cast/media_player.py", line 717, in async_play_media
    await self.hass.async_add_executor_job(
  File "/usr/local/lib/python3.9/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/usr/local/lib/python3.9/site-packages/pychromecast/quick_play.py", line 77, in quick_play
    controller.quick_play(**data)
  File "/usr/local/lib/python3.9/site-packages/pychromecast/controllers/media.py", line 544, in quick_play
    raise PyChromecastError()
pychromecast.error.PyChromecastError
2022-05-31 13:38:00 ERROR (MainThread) [homeassistant.core] Error executing service: <ServiceCall media_player.play_media (c:01G4CYGT8NMC639PQSYNPRFAKG): entity_id=['media_player.googlehome_library'], extra=enqueue=True, media_type=music, media_id=http://192.168.1.71:8095/media_player.googlehome_library.flac, enqueue=add>
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/core.py", line 1724, in catch_exceptions
    await coro_or_task
  File "/usr/src/homeassistant/homeassistant/core.py", line 1743, in _execute_service
    await cast(Callable[[ServiceCall], Awaitable[None]], handler.job.target)(
  File "/usr/src/homeassistant/homeassistant/helpers/entity_component.py", line 204, in handle_service
    await service.entity_service_call(
  File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 680, in entity_service_call
    future.result()  # pop exception if have
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 964, in async_request_call
    await coro
  File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 717, in _handle_entity_call
    await result
  File "/usr/src/homeassistant/homeassistant/components/cast/media_player.py", line 717, in async_play_media
    await self.hass.async_add_executor_job(
  File "/usr/local/lib/python3.9/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/usr/local/lib/python3.9/site-packages/pychromecast/quick_play.py", line 77, in quick_play
    controller.quick_play(**data)
  File "/usr/local/lib/python3.9/site-packages/pychromecast/controllers/media.py", line 544, in quick_play
    raise PyChromecastError()
pychromecast.error.PyChromecastError
2022-05-31 13:38:07 WARNING (MainThread) [homeassistant.components.media_player] Playing media with enqueue set to True is deprecated. Use 'add' instead
2022-05-31 13:38:07 WARNING (MainThread) [homeassistant.components.media_player] Playing media with enqueue set to True is deprecated. Use 'add' instead

[snip]

2022-05-31 13:38:20 WARNING (MainThread) [homeassistant.components.media_player] Playing media with enqueue set to True is deprecated. Use 'add' instead
2022-05-31 13:38:20 WARNING (MainThread) [homeassistant.components.media_player] Playing media with enqueue set to True is deprecated. Use 'add' instead
2022-05-31 13:38:20 ERROR (Thread-8) [pychromecast.socket_client] [Library(192.168.1.47):8009] Unhandled exception in worker thread, attempting reconnect
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/pychromecast/socket_client.py", line 538, in run
    if self.run_once(timeout=POLL_TIME_BLOCKING) == 1:
  File "/usr/local/lib/python3.9/site-packages/pychromecast/socket_client.py", line 632, in run_once
    self._route_message(message, data)
  File "/usr/local/lib/python3.9/site-packages/pychromecast/socket_client.py", line 712, in _route_message
    for handler in self._handlers[message.namespace]:
RuntimeError: Set changed size during iteration
2022-05-31 13:38:20 WARNING (Thread-8) [pychromecast.socket_client] [Library(192.168.1.47):8009] Error communicating with socket, resetting connection

[snip]

2022-05-31 13:38:21 WARNING (Thread-8) [pychromecast.socket_client] [Library(192.168.1.47):8009] Error communicating with socket, resetting connection
```

and again after restart:

<img width="765" alt="Schermafbeelding 2022-05-31 om 13 47 00" src="https://user-images.githubusercontent.com/33354141/171166068-23d3dcc5-4687-4aaa-8203-da6cffa67bbe.png">

**Comment by Mariusthvdb** [comment]
Still seeing this on .20

Logger: homeassistant.components.media_player
Source: components/media_player/__init__.py:376 
Integration: Media player (documentation, issues) 
First occurred: 22:16:11 (1 occurrences) 
Last logged: 22:16:11

Playing media with enqueue set to True is deprecated. Use 'add' instead

**Comment by Mariusthvdb** [comment]
Also, when moving the volume slider:

Logger: homeassistant.core
Source: components/cast/media_player.py:502 
First occurred: 22:20:11 (1 occurrences) 
Last logged: 22:20:11

Error executing service: <ServiceCall media_player.volume_set (c:01G4DWD8CXDSZEV2F58AQSSFP0): entity_id=['media_player.googlehome_library'], volume=0.42>
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/core.py", line 1724, in catch_exceptions
    await coro_or_task
  File "/usr/src/homeassistant/homeassistant/core.py", line 1743, in _execute_service
    await cast(Callable[[ServiceCall], Awaitable[None]], handler.job.target)(
  File "/usr/src/homeassistant/homeassistant/helpers/entity_component.py", line 204, in handle_service
    await service.entity_service_call(
  File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 680, in entity_service_call
    future.result()  # pop exception if have
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 964, in async_request_call
    await coro
  File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 717, in _handle_entity_call
    await result
  File "/usr/src/homeassistant/homeassistant/components/media_player/__init__.py", line 710, in async_set_volume_level
    await self.hass.async_add_executor_job(self.set_volume_level, volume)
  File "/usr/local/lib/python3.9/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/usr/src/homeassistant/homeassistant/components/cast/media_player.py", line 502, in set_volume_level
    self._chromecast.set_volume(volume)
  File "/usr/local/lib/python3.9/site-packages/pychromecast/controllers/receiver.py", line 181, in set_volume
    self.send_message({MESSAGE_TYPE: "SET_VOLUME", "volume": {"level": volume}})
  File "/usr/local/lib/python3.9/site-packages/pychromecast/controllers/__init__.py", line 121, in send_message
    self.send_message_nocheck(
  File "/usr/local/lib/python3.9/site-packages/pychromecast/controllers/__init__.py", line 133, in send_message_nocheck
    self._message_func(
  File "/usr/local/lib/python3.9/site-packages/pychromecast/socket_client.py", line 938, in send_platform_message
    return self.send_message(
  File "/usr/local/lib/python3.9/site-packages/pychromecast/socket_client.py", line 869, in send_message
    self._ensure_channel_connected(destination_id)
  File "/usr/local/lib/python3.9/site-packages/pychromecast/socket_client.py", line 982, in _ensure_channel_connected
    self.send_message(
  File "/usr/local/lib/python3.9/site-packages/pychromecast/socket_client.py", line 927, in send_message
    raise NotConnected(f"Chromecast {self.host}:{self.port} is connecting...")
pychromecast.error.NotConnected: Chromecast 192.168.1.47:8009 is connecting...

**Comment by marcelveldt** [comment]
1.1.20 is still not compatible with HA 2022.6

**Comment by Mariusthvdb** [comment]
O was I too hasty? I thought I saw the changes mention the fickle for the ‘add’

**Comment by marcelveldt** [comment]
haha, yes too hasty... I needed to do another patch release for the 2022.5 compatible version.
I'm now working on all changes for 2022.6 compatibility

**Comment by marcelveldt** [comment]
Released version 2022.6.b0 as prerelease for the "official" release version 2022.6
this one includes all changes for HA 2022.6
